modules/ui/osmose_details.js

```diff
@@ -0,0 +1,199 @@
+        if (issueString(_error, 'fix') !== unknown) {
+            let div = detailsEnter
+                .append('div')
+                .attr('class', 'error-details-subsection');
+
+            div.append('h4')
+                .text(() => t('QA.osmose.fix_title'));
+
+            div.append('p')
+                .html(d => issueString(d, 'fix'));
+        }
+
+        // Common Pitfalls (musn't exist for every issue type)
+        if (issueString(_error, 'trap') !== unknown) {
+            let div = detailsEnter
+                .append('div')
+                .attr('class', 'error-details-subsection');
+
+            div.append('h4')
+                .text(() => t('QA.osmose.trap_title'));
+
+            div.append('p')
+                .html(d => issueString(d, 'trap'));
+        }
+
+        detailsEnter
+            .append('div')
+            .attr('class', 'translation-link')
+            .append('a')
+            .attr('target', '_blank')
+            .attr('rel', 'noopener noreferrer') // security measure
+            .attr('href', 'https://www.transifex.com/openstreetmap-france/osmose')
+            .text(d => t('QA.osmose.translation'))
+            .append('svg')
+            .attr('class', 'icon inline')
+            .append('use')
+            .attr('href', '#iD-icon-out-link');
+
+        services.osmose.loadErrorDetail(_error, (err, d) => {
+            // No details to add if there are no associated issue elements
+            if (!d.elems || d.elems.length === 0) return;
+
+            // Things like keys and values are dynamically added to a subtitle string
+            if (d.detail) {
+                detailsDiv
+                    .append('h4')
+                    .attr('class', 'error-details-subtitle')
+                    .text(() => t('QA.osmose.detail_title'));
+
+                detailsDiv
+                    .append('p')
+                    .html(d => d.detail);
+            }
+
+            // Create list of linked issue elements
+            elemsDiv
+                .append('h4')
+                .attr('class', 'error-details-subtitle')
+                .text(() => t('QA.osmose.elems_title'));
+
+            elemsDiv
+                .append('ul')
+                .attr('class', 'error-details-elements')
+                .selectAll('.error_entity_link')
+                .data(d.elems)
+                .enter()
+                .append('li')
+                .append('a')
+                .attr('class', 'error_entity_link')
+                .text(d => d)
+                .each(function() {
+                    var link = d3_select(this);
+                    var entityID = this.textContent;
+                    var entity = context.hasEntity(entityID);
+
+                    // Add click handler
+                    link
+                        .on('mouseenter', () => {
+                            context.surface().selectAll(utilEntityOrMemberSelector([entityID], context.graph()))
```

Comment:
For consistency, we might try using the blue highlight instead of the red for this and the other issue types but we can change them later.

Reply:
Yeah there are a few behaviours such as this which are shared between the three QA layers which should probably all be changed together (tried to avoid differing in this PR for that reason)

Reply:
I figured. This release will be a good opportunity to suss some of these problems out.